src/texture/tex_impl.rs

```diff
@@ -264,3 +275,215 @@ impl Drop for TextureImplementation {
         });
     }
 }
+
+/// Checks that the texture format is supported and compatible with the client format.
+///
+/// Returns a GLenum suitable for the internal format of `glTexImage#D` and `glTexStorage*D`.
+fn format_request_to_glenum(display: &Display, client: ClientFormat, format: TextureFormatRequest)
+                            -> gl::types::GLenum
+{
+    let version = display.context.context.get_version();
+    let extensions = display.context.context.get_extensions();
+
+    match format {
+        TextureFormatRequest::AnyFloatingPoint => {
+            let size = client.get_num_components();
+
+            if version >= &GlVersion(3, 0) {
+                match size {
+                    1 => gl::RED,
+                    2 => gl::RG,
+                    3 => gl::RGB,
+                    4 => gl::RGBA,
+                    _ => unreachable!(),
+                }
+
+            } else {
+                size as gl::types::GLenum
+            }
+        },
+
+        TextureFormatRequest::Specific(TextureFormat::UncompressedFloat(f)) => {
+            assert!(version >= &GlVersion(3, 0));       // FIXME: 
```

Comment:
Will need to change this before merging. The `new_empty` functions fail this assertion.